### PR TITLE
arm64 kexec: separate testable code

### PR DESCRIPTION
**Pull request by hugelgupf**
Makes https://github.com/u-root/u-root/pull/2884 easier to review once merged. Simple code move.

### CODE REVIEW

File: pkg/boot/linux/load_linux_image.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	kernelAlignSize = 1 << 21 // 2 MB.
+)
+
+func mmap(f *os.File) (data []byte, ummap func() error, err error) {
+	s, err := f.Stat()
+	if err != nil {
+		return nil, nil, fmt.Errorf("stat error: %w", err)
+	}
+	if s.Size() == 0 {
+		return nil, nil, fmt.Errorf("cannot mmap zero-len file")
```
Comment (rminnich) [inline pkg/boot/linux/load_linux_image.go:45]: you could wrap os.ErrInvalid here maybe?

File: pkg/boot/linux/load_linux_image.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+	}
+	d, err := unix.Mmap(int(f.Fd()), 0, int(s.Size()), syscall.PROT_READ, syscall.MAP_PRIVATE)
+	if err != nil {
+		return nil, nil, fmt.Errorf("mmap failed: %w", err)
+	}
+
+	ummap = func() error {
+		if err := unix.Munmap(d); err != nil {
+			return fmt.Errorf("failed to unmap %s: %v", f.Name(), err)
```
Comment (rminnich) [inline pkg/boot/linux/load_linux_image.go:54]: Should this have %w?

File: pkg/boot/linux/load_linux_image.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+	return d, ummap, nil
+}
+
+// sanitizeFDT cleanups boot param properties from chosen node of the given FDT.
+func sanitizeFDT(fdt *dt.FDT) (*dt.Node, error) {
+	// Clear old entries in case we've already been through kexec to get
+	// to this instance of runtime.
+	chosen, _ := fdt.NodeByName("chosen")
+	if chosen == nil {
+		return nil, fmt.Errorf("no /chosen node in device tree")
```
Comment (rminnich) [inline pkg/boot/linux/load_linux_image.go:68]: %w with os.ErrNotExist? might make the test easier.

File: pkg/boot/linux/load_linux_image.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+		chosen.RemoveProperty(property)
+	}
+
+	return chosen, nil
+}
+
+func kexecLoadImage(kernel, ramfs *os.File, cmdline string, opts KexecOptions) (*kimage, error) {
+	fdt, err := dt.LoadFDT(opts.DTB)
+	if err != nil {
+		return nil, fmt.Errorf("loadFDT(%s) = %v", opts.DTB, err)
```
Comment (rminnich) [inline pkg/boot/linux/load_linux_image.go:80]: I may be missing something but ISTM a lot of these %v could be %w, in case we can get to tests later.

**Review by rminnich — COMMENTED** [review]
seems fine but I'm wondering if it is possible to wrap errors and in one case add os.ErrInvalid to faciliate tests at some point.

**Comment by hugelgupf** [comment]
I direct your attention to https://github.com/u-root/u-root/pull/2884 for the error wrapping and testing

**Pull request closed without merging**